### ISSUE TITLE
Internal Error when saving new selector on Views tab for a Test.

### ISSUE THREAD

**Issue by whitingjr** [Hyperfoil/Horreum#66]
When updating a Test to choose a selector on the Views tab I am seeing this error in the UI.

```
Warning alert:500 Internal Server Error

{
  "details": "javax.persistence.PersistenceException: org.hibernate.exception.DataException: could not execute statementError id 3dc24a58-f9d8-47d5-a358-10ef3e0133b1-18",
  "stack": ""
}
```

on the server this is reported in the logs

```
Caused by: org.postgresql.util.PSQLException: ERROR: jsonpath array subscript is not a single numeric value
  Where: SQL statement "WITH vcs AS (
                    SELECT id, unnest(regexp_split_to_array(accessors, ';')) as aa
                    FROM viewcomponent
                    WHERE id = NEW.id
                )
                INSERT INTO view_data
                SELECT vcs.id as vcid, rs.runid, array_agg(se.id) as extractor_ids,
                    jsonb_object_agg(se.accessor, (CASE
                        WHEN aa like '%[]' THEN jsonb_path_query_array(run.data, (rs.prefix || se.jsonpath)::jsonpath)
                        ELSE jsonb_path_query_first(run.data, (rs.prefix || se.jsonpath)::jsonpath)
                    END)) as object
                FROM vcs
                JOIN schemaextractor se ON se.accessor = replace(aa, '[]', '')
                JOIN run_schemas rs ON rs.schemaid = se.schema_id
                JOIN run on run.id = rs.runid
                GROUP BY runid, vcid"
PL/pgSQL function vd_after_update_vc_func() line 3 at SQL statement
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:481)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:401)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164)
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:130)
	at io.agroal.pool.wrapper.PreparedStatementWrapper.executeUpdate(PreparedStatementWrapper.java:88)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
	... 116 more


```

Implementation-Version: 0.1-SNAPSHOT

**Comment by johnaohara** [comment]
@whitingjr is this issue fixed?

**Comment by whitingjr** [comment]
@johnaohara yes

**Comment by rvansa** [comment]
Was not fixed, reopening.

**Comment by whitingjr** [comment]
my misunderstanding. The error no longer occurs for me.

**Comment by rvansa** [comment]
Fixed in 90cfd3b5e5e57db3475413aada50d12c18ad6fb6
